File: posterior_database/models/turing/earn_height/earn_height.jl
using Turing, LinearAlgebra

@model function earn_height(N, earn, height)
  beta ~ filldist(Turing.Flat(), 2)
  sigma ~ Turing.FlatPos(0)
  
  earn ~ MvNormal(beta[1] .+ beta[2] .* height, sigma^2*I)
end

File: posterior_database/models/pymc/earn_height.py
import pymc as pm
import pytensor.tensor as pt
import numpy as np

def make_model(data: dict, prior_only: bool = False) -> pm.Model:

    with pm.Model() as model:
        N = data['N']
        earn = data['earn']
        height = data['height']
        
        beta = pm.Flat("beta", shape=2)
        sigma = pm.HalfFlat("sigma")
        
        mu = pm.Deterministic("mu", beta[0] + beta[1] * height)
        
        if not prior_only:
            pm.Normal("earn", mu=mu, sigma=sigma, observed=earn)

    return model